### ISSUE TITLE
Webadmin authentication form does not accept password managers' credentials

### ISSUE THREAD

**Issue by nicofrand** [YunoHost/issues#2494]
### Describe the bug

I am using Bitwarden (not self-hosted) to store my credentials.
On Yunohost's authentication form, with Firefox for Android, when I select the credentials proposed by Bitwarden, then click on the submit button, the login field is emptied and an error states that it must not be empty.

If I select the login field, erase the last char and type it again, everything works. Which makes me think this is an issue of paste management in the webui.

**This is a new issue, that was not happening before Yunohost 12**.

### Context

- Hardware: *VPS bought online*
- YunoHost version: 12.0.7
- I have access to my server: *Through SSH | through the webadmin | direct access via keyboard / screen | ...*

### To reproduce

* Use Firefox for Android (I did not try another browser)
* Use Bitwarden
* On the Yunohost's auth form, use Bitwarden to fill the fields
* Click on the submit button

### Expected behavior

The form submission works as expected, without having to manually edit the fields.

**Comment by OniriCorpe** [comment]
hi

is your browser up to date?

three person from the team have tried to reproduce without success...

**Comment by nicofrand** [comment]
Hi!

It is, I update it almost daily (I am using the beta version, but the standard version has the same issue).
I'll see if there are more steps to it…

**Comment by Salamandar** [comment]
My only idea is that your login URL in bitwarden does not match the web page's URL.

**Comment by nicofrand** [comment]
There is a misunderstanding: I am able to fill the fields with Bitwarden. The issue is that the form does not think the fields have been filled when I submit the form, it acts like the fields are empty.

**Comment by kay0u** [comment]
I do have the same issue on my Android phone (Firefox browser) on the admin login form (not the user portal).

As a workaround for now, I click on the password field, then on the user field, add and remove a character, then I can press the login button.

**Comment by Salamandar** [comment]
> I do have the same issue on my Android phone (Firefox browser) on the admin login form (not the user portal).
> 
> As a workaround for now, I click on the password field, then on the user field, add and remove a character, then I can press the login button.

Went here to say that. @nicofrand can you confirm we are talking about admin and not sso ?

**Comment by nicofrand** [comment]
Indeed, sorry I mixed it up!

Also, it somehow seems to happen more when focusing the password field than the username field...

**Comment by m4lvin** [comment]
I can confirm this, and it is not bitwarden specific as I use https://codeberg.org/PassFF/passff

It does manage to fill the form, but when I click "Login" it says "The form contains some errors." and the username field is marked red, even though it contains the correct user name. If I then click into the username field, add a single space or other character and delete that again, then I can click "Login" and it works now.

I suggest to rename the issue to something more general like "Admin Authentication form does not work when filled automatically wth a password manager" or so.

My blind guess for the underlying reason would be that somehow the form being defined in Vue Script does not update its internal(?) values and does not notice the form has been filled when this is done by a browser extension instead of keypresses by the user?

Is this the place to look?

https://github.com/YunoHost/yunohost-admin/blob/63f4cfb6df44819dd0619dafa7662aff6460fae9/app/src/views/LoginView.vue#L25-L53

**Comment by kay0u** [comment]
I use a remote debugger to see what was the issue on my android:

As you can see in these screenshots, the `value` of the password field is empty in the inspect tab, although it does contain a password (you have to trust me, because I blur it for obvious reason)

![Image](https://github.com/user-attachments/assets/068e5414-79cf-403d-8189-076e0c77cc39)

![Image](https://github.com/user-attachments/assets/573fc2b4-ff99-41ea-9ace-dbdaba4231c4)

I'm not sure why the [`LoginView` in `yunohost-admin`](https://github.com/YunoHost/yunohost-admin/blob/63f4cfb6df44819dd0619dafa7662aff6460fae9/app/src/views/LoginView.vue#L25-L53) and [the one in `yunohost-portal`](https://github.com/YunoHost/yunohost-portal/blob/75dc549e75750790ac3514c13117fac8ab1afbfe/pages/login.vue#L21-L28) are so different, it does work well on the portal version

**Comment by nicofrand** [comment]
I think there is a difference between the HTML and the DOM: the attribute can be empty/non-existing but the `input.value` will be filled.

**Comment by kay0u** [comment]
Sometimes, it's the username which is empty, and the password `value` is correctly set

Edit:

> I think there is a difference between the HTML and the DOM: the attribute can be empty/non-existing but the input.value will be filled.

When the value is empty in the HTML and I click the login button, I get the error message "empty password/login"

**Comment by guidovw** [comment]
Same error with ProtonPass. What works for me is I click in the password field and let it fill in the credentials a second time. Then there is no error.

**Comment by OniriCorpe** [comment]
Interestingly I don’t have any issue on my side, using the Bitwarden app integration with Safari on iOS / iPadOS / macOS or the Bitwarden  extension on Firefox / Zen on EndeavourOS (Arch Linux)

Using YunoHost-admin 12.0.12

What I understand from the comments is that it’s only happening with Firefox for android, isn’t it?
If so, we should rename the issue to specify it

**Comment by chri2** [comment]
> it’s only happening with Firefox for android, isn’t it?

Nope. Firefox 136.0.1 (64-Bit) on Debian Bookworm + PassFF 1.21 is the same.

**Comment by guidovw** [comment]
Here Brave 1.76.74, both on macOS Sequoia 15.3.2 and Debian Bookworm

**Comment by nicofrand** [comment]
FWIW I use the Bitwarden app, not the Firefox's extension.

**Comment by brimwats** [comment]
I see this with Proton Pass and Firefox Android/Zen Browser (based on firefox) on desktop.

On Chrome (desktop) and Vivaldi (android and desktop) I do not have this issue. The common factor for me is firefox.

Secondly, it's always the username, never the password.

**Comment by DeMiro5001** [comment]
I had a 'similar' issue on chrome, but that was on yunohost 11, I noticed that it was fixed by waiting a little bit after the login page has fully loaded before filling credentials. But this doesn't seem to be the same issue.
Anyway, I found an issue on vue.js that discusses it with some workarounds, hope this helps https://github.com/vuejs/vue/issues/1331

**Comment by OniriCorpe** [comment]
> I see this with Proton Pass and Firefox Android/Zen Browser (based on firefox) on desktop.
> 
> On Chrome (desktop) and Vivaldi (android and desktop) I do not have this issue. The common factor for me is firefox.


On which desktop OS? Either endeavorOS (arch) I don’t have this issue at all
